File: app/src/main/java/xyz/farshad/vocab/activity/LevelListActivity.java
package xyz.farshad.vocab.activity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import com.orm.SugarContext;

import java.util.List;

import androidx.appcompat.app.AppCompatActivity;
import xyz.farshad.vocab.R;
import xyz.farshad.vocab.component.DataAdapter.LevelListAdapter;
import xyz.farshad.vocab.model.Level;

public class LevelListActivity extends AppCompatActivity {

    List<Level> levels;
    private Long courseId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_level_list);
        setTitle("Levels");
        SugarContext.init(LevelListActivity.this);

        Bundle b = getIntent().getExtras();
        if(b != null && b.containsKey("courseId")){
            courseId = b.getLong("courseId");
            levels = Level.find(Level.class, "course_id = ?", courseId.toString());

            showLevelList(false);
        }
    }

    private void showLevelList(boolean reload){
        if (reload){
            levels = Level.listAll(Level.class);
        }
        ArrayAdapter<Level> adapter = new LevelListAdapter(this, R.layout.level_list_view, levels);
        ListView list = findViewById(R.id.levelListView);
        list.setAdapter(adapter);
        list.setItemsCanFocus(true);
    }
}

File: app/src/main/java/xyz/farshad/vocab/activity/LevelListActivity.kt
package xyz.farshad.vocab.activity

import android.os.Bundle
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import org.koin.androidx.viewmodel.ext.android.viewModel
import xyz.farshad.vocab.R
import xyz.farshad.vocab.component.DataAdapter.LevelListAdapter
import xyz.farshad.vocab.data.model.Level
import xyz.farshad.vocab.viewmodel.LevelViewModel

class LevelListActivity : AppCompatActivity() {
    private val levelViewModel: LevelViewModel by viewModel()
    private var courseId: Long? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_level_list)
        title = "Levels"

        val b = intent.extras
        if (b != null && b.containsKey("courseId")) {
            courseId = b.get("courseId").toString().toLong()
            levelViewModel.findByCourseId(courseId!!.toInt())
        }

        setObserver()
    }

    private fun setObserver() {
        levelViewModel.watchLevel()?.observe(this, {
            showLevelList(it)
        })
    }

    private fun showLevelList(levels: List<Level>) {
        val adapter = LevelListAdapter(this, R.layout.level_list_view, levels)
        val list = findViewById<ListView>(R.id.levelListView)
        list.adapter = adapter
        list.itemsCanFocus = true
    }
}

